stable/aws-vpc-cni/templates/clusterrole.yaml

```diff
@@ -8,17 +8,18 @@ rules:
   - apiGroups:
       - crd.k8s.amazonaws.com
     resources:
-      - "*"
-      - namespaces
-    verbs:
-      - "*"
+      - eniconfigs
+    verbs: ["list", "watch", "get"]
   - apiGroups: [""]
     resources:
       - pods
-      - nodes
       - namespaces
     verbs: ["list", "watch", "get"]
+  - apiGroups: [""]
+    resources:
+      - nodes
+    verbs: ["list", "watch", "get", "update"]
   - apiGroups: ["extensions"]
     resources:
-      - daemonsets
+      - '*'
```

Comment:
are there any specific resources?

Reply:
I believe this change was made to match the config in the CNI repo, [v1.7/aws-k8s-cni.yaml](https://github.com/aws/amazon-vpc-cni-k8s/blob/v1.7.3/config/v1.7/aws-k8s-cni.yaml#L46-L52). The change was added in https://github.com/aws/amazon-vpc-cni-k8s/pull/1082 ping @jayanthvn

Reply:
If we know for sure which resources are needed, we can update the helm chart independently of the cni repo. By specifying * we might be allowing more than necessary access.

Reply:
It has been a while since I've had to write policies and not entirely sure where to get the entire list outside of kubectl cache but here are the resources I came up with for the extensions api group:
```
DaemonSet
Deployment
DeploymentRollback
Ingress
NetworkPolicy
PodSecurityPolicy
ReplicaSet
Scale
```
I might go try to figure out what ipamd change sparked the new resource requirement from that pr.

Reply:
@Kishorb @mogren I wasn't able to figure out what specific resources inside extension would be required outside of DaemonSet though I do know there are deprecation changes post 1.16 to the extensions group. I see my changes proposed here as orthogonal to the granularity of clusterrole policies as they represent upstream definitions more closely related to the vpc cni project.  I do however agree with the least privileged approach and think this should be figured out rather soon. I'm not sure what other help I can provide, please let me know!

Reply:
@Kishorb I agree that we should scope this down, but I guess for now it would be ok that the Helm chart is matching the current CNI config sample. That said, I opened https://github.com/aws/amazon-vpc-cni-k8s/issues/1232 to address this in the CNI repo, and we should follow up with a PR here to match.

Reply:
sounds good.